spec/download_payment_batch_detail_report_spec.rb

```diff
@@ -0,0 +1,34 @@
+# frozen_string_literal: true
+
+require 'spec_helper'
+require_relative '../CyberSource/download_payment_batch_detail_report'
+
+RSpec.describe DownloadPaymentBatchDetailReport do
+  let(:result) { described_class }
+  let(:download_report) { File.open(File.join(Dir.pwd, 'spec', 'fixtures', 'report_file.csv')).read }
+  let(:d) { Date.today }
+  let(:date_stamp) { Date.new(d.year, d.month - 1, 1).strftime('%Y%m') }
+  let(:file) { File.join(Dir.pwd, 'spec', 'fixtures', "report_cyberfile.#{date_stamp}") }
+
+  before do
+    File.open(file, 'a+')
+    allow_any_instance_of(result).to receive(:download_report).and_return(download_report)
+    result.new.main
+  end
+
+  after do
+    File.delete(file) if File.exist?(file)
+  end
+
+  it 'downloads reports for the entire month' do
+    puts "Opening file: #{file}\n"
+    expect(IO.readlines(file).size).to be >= 28
+  end
+
+  it 'contains a csv file of details about the transactions' do
+    puts "Opening file: #{file}\n"
+    IO.foreach(file) do |row|
+      expect(row.split(',').size).to eq 13
+    end
+  end
```

Comment:
When I run this locally, and also look at the Travis build. I notice that the output of rspec is list of dates. Is this intended?

Reply:
Yes, that is because the ksh script that calls and consumes this data expects to write the stdout to a log file, so when the test runs, it also outputs the stdout.